### PR TITLE
 [ORBIT-9793] Auto deployment for Lambda

### PR DESCRIPTION
**Pull request by ng-htec**
add: artifacts-ci-lambda.yml
add: lambda-deploy.yml

### CODE REVIEW

File: .github/workflows/lambda-deploy.yml [hunk 1]
```diff
@@ -3,71 +3,65 @@ name: Deploy Lambda
 on:
```
Comment (jzlatanovic) [inline .github/workflows/lambda-deploy.yml:3]: I think you don't need this file

File: .github/workflows/artifacts-ci-lambda.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          zip -qq -r -j ${{ steps.repo_data.outputs.app_version }}.zip ./dist/*
+      - name: upload lambda function
+        run: |
+            aws s3 cp ${{ steps.repo_data.outputs.app_version }}.zip s3://${{ inputs.s3_bucket }}/${{ inputs.s3_bucket_key }}/${{ inputs.app_name }}/
+      - name: Comment PR
+        id: comment_pr
+        if: github.event_name == 'pull_request'
+        uses: htec-infra/gha-utils/actions/comment-pr@v0.3.1
+        with:
+          github_token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (jzlatanovic) [inline .github/workflows/artifacts-ci-lambda.yml:99]: is it `github_token` or `gh_token`

File: .github/workflows/artifacts-ci-lambda.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    runs-on: [ubuntu-latest]
+    if: contains(needs.build_push.outputs.app_version, 'alpha')
+    steps:
+      - name: Trigger deploy
+        uses: htec-infra/gha-utils/actions/repo-dispatch@v0.3.1
+        with:
+          gh_token: ${{ secrets.gh_token }}
+          repository: ${{ inputs.infra_repo }}
+          event_type: ${{ inputs.event_type }}
+          client_payload: '{"env": "${{ inputs.deploy_environment }}", "name": "${{ inputs.app_name || steps.repo_data.outputs.repo_name }}", "s3_zip_file": "${{ needs.build_push.outputs.app_version }}.zip", "s3_bucket": "${{ inputs.s3_bucket }}", "dry_run": "${{ inputs.dry_run }}"}'
```
Comment (jzlatanovic) [inline .github/workflows/artifacts-ci-lambda.yml:115]: Do you need all this parameters?

Comment (ng-htec) [reply .github/workflows/artifacts-ci-lambda.yml:115]: Unfortunately yes, they are all necessary in order to update lambda function. Maybe we could move s3_bucket parameter to other place but then we will lose some of the flexibility

File: .github/workflows/artifacts-ci-lambda.yml (hunk 2 above)
Comment (ng-htec) [reply .github/workflows/artifacts-ci-lambda.yml:99]: At the moment it's github_token

**Review by jzlatanovic — APPROVED** [review]

**Pull request merged**